File: src/com/suushiemaniac/cubing/alglib/util/FixArrayComparator.java
package com.suushiemaniac.cubing.alglib.util;

import java.util.Comparator;

public class FixArrayComparator<T> implements Comparator<T> {
    private T[] preOrder;

    public FixArrayComparator(T[] preOrder) {
        this.preOrder = preOrder;
    }

    @Override
    public int compare(T o1, T o2) {
        return Integer.compare(ArrayUtils.arrayIndex(this.preOrder, o1), ArrayUtils.arrayIndex(this.preOrder, o2));
    }
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/util/FixArrayComparator.kt
package com.suushiemaniac.cubing.alglib.util

class FixArrayComparator<T>(private val preOrder: Array<T>) : Comparator<T> {
    override fun compare(o1: T, o2: T): Int {
        return Integer.compare(this.preOrder.indexOf(o1), this.preOrder.indexOf(o2))
    }
}
